DungeonMaster.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <vector>
+#include <iostream>
+#include "Unit.h"
+
+class DungeonMaster
+{
+private:
+    static DungeonMaster *instance;
+    std::vector<Unit *> units;
+
+    DungeonMaster();
+
+public:
+    static DungeonMaster *GetInstance();
+    void Init(char *argv[]);
+    ~DungeonMaster();
+
+    void Fight(Unit *atk, Unit *def);
+};
```

Comment:
Ez manager osztaly also hangon overkill, de amugy miert hasznalod a singleton patternt?

Reply:
a DungeonMaster-t azért gondoltuk célszerűnek, mert ha ez egy terebélyes rpg játék lesz félév végére, akkor valószínűleg lesz sok olyan funckionalitás, ami nem köthető majd egy Unit-hoz. A feladatok elszeparáltsága volt vele a cél. Azért singleton, hogy véletlenül se lehessen 2-t létrehozni. Töröljük ezt az osztályt?

Reply:
@hegyhati

Reply:
Kapok, kapok :-)  Alapvetoen nem buta dolog neha kicsit elore gondolkodni, ha tudod, hogy mi lesz. Ha nem, akkor jo esellyel csinalsz olyat, ami felesleges lesz, mert nem abba az iranyba megy az egesz, mint kellene.  Jelenleg egyetlen oda-vissza csapkodas a Unit fuggetlen logika, ami kb 1 for meg 2 if, ami elfer egy main-ben kb 6 sorban. De felolem maradhat egy ilyen manager osztaly. szerintem Nektek lesz tobb szenvedes vele kesobb, de akkor legalabb ne szemeteljen :-)

Reply:
@hegyhati  elvégeztük a kért változtatásokat